runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -0,0 +1,2505 @@
+						T::FudgeHandle::SIBLING_ID
+					)),
+					transfer_amount
+				);
+			});
+
+			env.pass(Blocks::ByNumber(1));
+
+			env.sibling_state_mut(|| {
+				let current_balance =
+					orml_tokens::Pallet::<T>::free_balance(cfg_in_sibling, &Keyring::Bob.into());
+
+				// Verify that Keyring::Bob now has (amount transferred - fee)
+				assert_eq!(current_balance, transfer_amount - fee(18));
+
+				// Sanity check for the actual amount Keyring::Bob ends up with
+				assert_eq!(current_balance, 4992960800000000000);
+			});
+		}
+
+		fn test_cfg_transfers_to_and_from_sibling<T: Runtime + FudgeSupport>() {
+			let mut env = FudgeEnv::<T>::from_storage(
+				Genesis::default()
+					.add(genesis::balances::<T>(cfg(10)))
+					.storage(),
+				Default::default(),
+			);
+
+			setup_xcm(&mut env);
+
+			// In order to be able to transfer CFG from Sibling to Centrifuge, we need to
+			// first send CFG from Centrifuge to Sibling, or else it fails since it'd be
+			// like Sibling had minted CFG on their side.
+			transfer_cfg_to_sibling(&mut env);
+
+			let alice_initial_balance = cfg(5);
+			let bob_initial_balance = cfg(5) - cfg_fee();
+			let transfer_amount = cfg(1);
+			// Note: This asset was registered in `transfer_cfg_to_sibling`
+			let cfg_in_sibling = CurrencyId::ForeignAsset(12);
+
+			env.parachain_state_mut(|| {
+				assert_eq!(
+					pallet_balances::Pallet::<T>::free_balance(&Keyring::Alice.into()),
+					alice_initial_balance
+				);
+			});
+
+			env.sibling_state_mut(|| {
+				assert_eq!(
+					pallet_balances::Pallet::<T>::free_balance(&parachain_account(
+						T::FudgeHandle::PARA_ID
+					)),
+					0
+				);
+				assert_eq!(
+					orml_tokens::Pallet::<T>::free_balance(cfg_in_sibling, &Keyring::Bob.into()),
+					bob_initial_balance
+				);
+			});
+
+			env.sibling_state_mut(|| {
+				assert_ok!(orml_xtokens::Pallet::<T>::transfer(
+					RawOrigin::Signed(Keyring::Bob.into()).into(),
+					cfg_in_sibling,
+					transfer_amount,
+					Box::new(
+						MultiLocation::new(
+							1,
+							X2(
+								Parachain(T::FudgeHandle::PARA_ID),
+								Junction::AccountId32 {
+									network: None,
+									id: Keyring::Alice.into(),
+								}
+							)
+						)
+						.into()
+					),
+					WeightLimit::Limited(8_000_000_000_000.into()),
+				));
+
+				// Confirm that Bobs's balance is initial balance - amount transferred
+				assert_eq!(
+					orml_tokens::Pallet::<T>::free_balance(cfg_in_sibling, &Keyring::Bob.into()),
+					bob_initial_balance - transfer_amount
+				);
+			});
+
+			env.pass(Blocks::ByNumber(2));
+
+			env.parachain_state_mut(|| {
+				// Verify that Keyring::Alice now has initial balance + amount transferred - fee
+				assert_eq!(
+					pallet_balances::Pallet::<T>::free_balance(&Keyring::Alice.into()),
+					alice_initial_balance + transfer_amount - cfg_fee(),
+				);
+			});
+		}
+
+		fn transfer_ausd_to_centrifuge<T: Runtime + FudgeSupport>() {
```

Comment:
@cdamian how much work would it be to have another parachain that acts as the reserve for USDC (i.e. AssetHub) and test transferring in and out from that?

Reply:
Assuming you want to also use the `AssetHub` runtime right? We would need to change the fudge env to provide a builder for that and then use it as a para. Furthermore, adding one more parachain would make block building even slower, imo that's something that we should consider.

Reply:
Okay. Lets keep it for now.